### ISSUE TITLE
New PostgreSQL Module for native apps

### ISSUE THREAD

**Issue by oleg-nenashev** [testcontainers/testcontainers-native#41]
Many C++ applications utilize PostgreSQL in the testing pipelines, it would be nice to have a Testcontainers module for it that simplifies the usage. Since there is a Golang implementation of a module, it might be also a good use-case of creating bridge APIs with Cgo, though a module could be a very simple C library

## References

- https://testcontainers.com/modules/postgresql/

**Comment by SapiensAnatis** [comment]
I would be interested in contributing this as I have a use for it. I've just tried to start it as a generic container, but I think we are missing several APIs:

- a way to set environment variables (for `POSTGRES_PASSWORD` which is required for the image to start)
- a way to wait on something other than a HTTP port (either for the TCP port to start listening, or even better `pg_isready`)

Should adding these be tackled under separate issues? I could not find an existing issue for either one.

**Comment by oleg-nenashev** [comment]
@SapiensAnatis having separate issues would be nice but not necessary. If you submit changes in a single patch, I can get them landed and have changelings polished manually
